lib/remote-settings.js

```diff
@@ -0,0 +1,51 @@
+"use strict";
+
+const got = require("got");
+const AppConstants = require("../app-constants");
+
+
+const BREACHES_COLLECTION = "fxmonitor-breaches";
+const FX_RS_COLLECTION = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-workspace/collections/${BREACHES_COLLECTION}`;
+const FX_RS_RECORDS = `${FX_RS_COLLECTION}/records`;
+const FX_RS_BEARER_TOKEN = AppConstants.FX_REMOTE_SETTINGS_BEARER_TOKEN;
+
+const RemoteSettings = {
+
+  async whichBreachesAreNotInRemoteSettingsYet(breaches) {
+    const fxRSRecords = await got(FX_RS_RECORDS, {
+      json: true,
+      headers: {
+        "authorization": `Bearer ${FX_RS_BEARER_TOKEN}`,
+      },
+    });
+    const remoteSettingsBreachesSet = new Set(
+      fxRSRecords.body.data.map(b => b.Name)
+    );
+
+    return breaches.filter( ({Name}) => !remoteSettingsBreachesSet.has(Name) );
+  },
+
+  async postNewBreachToBreachesCollection(data) {
+    // Create the record
+    await got.post(FX_RS_RECORDS, {
+      headers: {
+        "Content-Type": "application/json",
+        "authorization": `Bearer ${FX_RS_BEARER_TOKEN}`,
+      },
```

Comment:
Not sure if it helps here or not, but it looks like we can create a **got** client by using [`got.extend()`](https://github.com/sindresorhus/got#gotextendoptions) and possibly get away w/ declaring the `Content-Type`, `Authorization` headers and `json: true` in a single place [maybe].

```js
const client = got.extend({
  json: true,
  headers: {
    "Content-Type": "application/json", // possibly unneeded if we define `json: true` above.
    "Authorization": `Bearer ${FX_RS_BEARER_TOKEN}`
  }
});

// 1
const fxRSRecords = await client(FX_RS_RECORDS);
const remoteSettingsBreachesSet = new Set(fxRSRecords.body.data.map(b => b.Name));

// 2
await client.post(FS_RS_RECORDS, {
  body: JSON.stringify({data})
});

// 3
await client.patch(FS_RS_COLLECTION, {
  body: JSON.stringify({data: {status: "to-review"}})
});
```

---

This worked locally, but I had to do some mild tweaking since I didn't have a bearer token or know the exact `FX_REMOTE_SETTINGS_WRITER_SERVER` URL. Not sure if we should add that to the .env-dist file or somethin'.

<details>
<summary>Expand</summary>

```js
const got = require("got");

const AppConstants = {
  FX_REMOTE_SETTINGS_WRITER_SERVER: "https://firefox.settings.services.mozilla.com/v1"
};
const BREACHES_COLLECTION = "fxmonitor-breaches";

const FX_RS_COLLECTION = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main/collections/${BREACHES_COLLECTION}`;
const FX_RS_RECORDS = `${FX_RS_COLLECTION}/records`;

const client = got.extend({
  json: true,
  headers: {
    "Content-Type": "application/json"
  }
});

main();

async function main() {
  const fxRSRecords = await client(FX_RS_RECORDS);
  const remoteSettingsBreachesSet = new Set(fxRSRecords.body.data.map(b => b.Name).sort());
  console.log(remoteSettingsBreachesSet);
}
```
</details>

Reply:
I filed https://github.com/mozilla/blurts-server/issues/859 for this. I'd like to keep the script as-is for merging, because I just ran it and it added the new Bukalapak breach perfectly.